### PR TITLE
Basic working version of having 2 renderers render the same scene

### PR DESCRIPTION
**Pull request by gero3**
This pull request adds the possibility to use multiple renderers for the same scene.

It uses a vault to identify which renderData needs to be used based on the object and the renderer. The vault has also support for flags, which mean that if something is flagged as needsupdate it is communicated to the other renderers. This is done in a asynchronyous way, so that it only updates when it is ever needed. 

image

### CODE REVIEW

**Comment by gero3** [comment]
image

A not so basic version.

**Comment by mrdoob** [comment]
Hold on. Hold on.

I was going to do something similar to [WebGLExtensions](https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLExtensions.js):

```javascript
THREE.WebGLRenderer = function () {
    var buffers = new THREE.Buffers( gl );
    // at render time
    var buffer = buffers.get( object );
};

THREE.Buffers = function ( gl ) {
    var buffers = {};
    this.get = function ( object ) {
        var hash = object.geometry.id + '_' + object.material.id;
        if ( buffers[ hash ] !== undefined ) {
            return buffers[ hash ];
        }
        // create buffers
    };
};

**Comment by gero3** [comment]
I could incorporate that in this design. 

A few points to note tough:

It does more then just keep buffers separate for each renderer.
For instance, custom attributes,uniforms, programs and textures are also kept seperate. It also has the advantage of being useful for other renderers as any sort of data can be stored in it. 

The flagging system is also another part that is interesting. It allows renderers to be check if it was a flag was set since the last time it checked. This means that you can have slower updates for some of the renderers compared to others without incurring a slow down on the fast renderer. The user on the other hand only has to set a boolean property on the object to set a flag.

It also doesn't require an intervention in user code. The only persons who need to keep it in mind are the ones writing new renderers if they want to play nice with the other renderers. 

To make the cloth render in 2 renderers, the only thing I changed in the example was to add a renderer. The following lines are the only ones I copied to add the new Webglrenderer:

- https://github.com/mrdoob/three.js/blob/master/examples/webgl_animation_cloth.html#L285-L295
- https://github.com/mrdoob/three.js/blob/master/examples/webgl_animation_cloth.html#L317
- https://github.com/mrdoob/three.js/blob/master/examples/webgl_animation_cloth.html#L369

So basically, adding a renderer with the same parameters but of course only half the width.

Ps: I'll give a more torough explantation about what happens internally soon.

**Comment by gero3** [comment]
[An example](https://preview.c9.io/gero3/testprojects/2Renderers/webgl_performance_2_Renderers.html)

**Comment by mrdoob** [comment]
Sorry @gero3. Ended up merging #6723 instead.
I hope you don't hate me too much 😇
